.github/workflows/build.yml

```diff
@@ -39,28 +39,25 @@ jobs:
       matrix:
         php: [ 8.1, 8.2, 8.3, 8.4 ]
         composer-flags: [ "" ]
-        symfony-version: [ "" ]
         include:
           - php: 8.1
-            symfony-version: '^5.4'
+            composer-flags: "--with=symfony/yaml:^5.4"
           - php: 8.1
-            symfony-version: '^6.4'
+            composer-flags: "--with=symfony/yaml:^6.4"
     steps:
       - uses: actions/checkout@v4
-
-      - name: Setup PHP
-        uses: shivammathur/setup-php@v2
+      - uses: shivammathur/setup-php@v2
         with:
           php-version: "${{ matrix.php }}"
           ini-file: "development"
-          coverage: none
-
-      - name: Update Symfony version
-        if: matrix.symfony-version != ''
-        run: composer require --no-update "symfony/symfony:${{ matrix.symfony-version }}"
-
-      - name: Install dependencies
-        run: composer update ${{ matrix.composer-flags }}
-
-      - name: Run tests (phpunit)
-        run: ./vendor/bin/phpunit
+          coverage: "xdebug"
+      - run: composer update ${{ matrix.composer-flags }} --ansi --no-progress
+      - run: composer test -- --coverage-clover=coverage.xml --log-junit junit.xml --testdox
+      - uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@uuf6429 I have authorised the codecov app (for Behat/Gherkin only at this point, we can add it to other repositories as and when required).

I'm still not clear why the workflow specifies a `secrets.CODECOV_TOKEN` that is not defined (and therefore empty) and whether that will continue to work once this PR is merged. I'm somewhat surprised if Codecov allow anyone to upload coverage reports unauthenticated, but maybe that's the case. Can you provide any more info on how this works? If we don't in fact require a secret then we should remove these `token` arguments from the yaml to avoid confusion.

Reply:
As I understand, there were 3 options:
1. per-repository secret
2. per-organisation secret
3. github app/plugin

so I'm guessing that the secret is no longer needed with the 3rd option. I'll remove it and we can see what happens. 🤷

Reply:
Seems to still be working fine. My guess is that:
- codecov already knows about this project (it even lists [other behat org projects](https://app.codecov.io/gh/Behat))
- codecov understands that this is a PR run (it even shows that in the branch dropdown, prefixing my branches with my username)

Reply:
I guess easiest way to see this is to get it merged (after the other point(s) are also considered).

@carlos-granados can you also give this PR a look please?